qemu/tests/discard_for_raw_block_target.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import os
+
+from provider.qemu_img_utils import strace
+
+from avocado import fail_on
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest.qemu_storage import QemuImg
+
+
+def run(test, params, env):
+    """
+    qemu-img supports 'discard' for raw block target images.
+
+    1. Create source image via dd with all zero.
+    2. Modprobe a 1G scsi_debug disk with writesame_16 mode.
+    3. Trace the system calls while converting the zero image to the
+        scsi_debug block device, then check whether 'fallocate' system call
+        works in the log.
+    :param test: Qemu test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def check_output(strace_output, match_str):
+        """Check whether the value is good in the output file."""
+        logging.debug("Check the output file '%s'" % strace_output)
+        with open(strace_output) as fd:
+            if match_str not in fd.read():
+                test.fail("The system call is not right, "
```

Comment:
failure should more specific like no `fallocate` syscall detected.

Reply:
Updated, thanks.